prow/cmd/hook/trigger.go

```diff
@@ -0,0 +1,199 @@
+			return nil
+		}
+
+		pr, err := ga.GitHubClient.GetPullRequest(owner, name, number)
+		if err != nil {
+			return err
+		}
+
+		for _, job := range requestedJobs {
+			kr := makeKubeRequest(job, *pr)
+			ga.BuildRequests <- kr
+		}
+	}
+	return nil
+}
+
+func (ga *GitHubAgent) buildAll(pr github.PullRequest) {
+	for _, job := range ga.JenkinsJobs.AllJobs(pr.Base.Repo.FullName) {
+		if !job.AlwaysRun {
+			continue
+		}
+		kr := makeKubeRequest(job, pr)
+		ga.BuildRequests <- kr
+	}
+}
+
+func (ga *GitHubAgent) deleteAll(pr github.PullRequest) {
+	for _, job := range ga.JenkinsJobs.AllJobs(pr.Base.Repo.FullName) {
+		kr := makeKubeRequest(job, pr)
+		ga.DeleteRequests <- kr
+	}
+}
+
+func makeKubeRequest(job JenkinsJob, pr github.PullRequest) KubeRequest {
+	return KubeRequest{
+		JobName: job.Name,
+		Context: job.Context,
+
+		RerunCommand: job.RerunCommand,
+
+		RepoOwner: pr.Base.Repo.Owner.Login,
+		RepoName:  pr.Base.Repo.Name,
+		PR:        pr.Number,
+		Author:    pr.User.Login,
+		BaseRef:   pr.Base.Ref,
+		BaseSHA:   pr.Base.SHA,
+		PullSHA:   pr.Head.SHA,
+	}
+}
+
+// trustedPullRequest returns whether or not the given PR should be tested.
+// It first checks if the author is in the org, then looks for "ok to test
+// comments by org members.
+func (ga *GitHubAgent) trustedPullRequest(pr github.PullRequest) (bool, error) {
+	author := pr.User.Login
+	// First check if the author is a member of the org.
+	orgMember, err := ga.isMember(author)
+	if err != nil {
+		return false, err
+	} else if orgMember {
+		return true, nil
+	}
+	// Next look for "ok to test" comments on the PR.
+	comments, err := ga.GitHubClient.ListIssueComments(pr.Base.Repo.Owner.Login, pr.Base.Repo.Name, pr.Number)
+	if err != nil {
+		return false, err
+	}
+	for _, comment := range comments {
+		commentAuthor := comment.User.Login
+		// Skip comments by the PR author.
+		if commentAuthor == author {
+			continue
+		}
+		// Skip bot comments.
+		if commentAuthor == "k8s-bot" || commentAuthor == "k8s-ci-robot" {
+			continue
+		}
+		// Look for "ok to test"
+		if !okToTest.MatchString(comment.Body) {
+			continue
+		}
+		// Ensure that the commenter is in the org.
+		commentAuthorMember, err := ga.isMember(commentAuthor)
+		if err != nil {
+			return false, err
+		} else if commentAuthorMember {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (ga *GitHubAgent) askToJoin(pr github.PullRequest) error {
+	commentTemplate := `
+Can a [%s](https://github.com/orgs/%s/people) member verify that this patch is reasonable to test? If so, please reply with "@k8s-bot ok to test" on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands will still work. Regular contributors should join the org to skip this step.
+
+If you have questions or suggestions related to this bot's behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra) repository.`
```

Comment:
maybe link to the new issue page with a template?
like this:

https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:&body=%0A%0D/assign%20@spxtr